exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/UnionAllPrule.java

```diff
@@ -61,7 +61,8 @@ public void onMatch(RelOptRuleCall call) {
         convertedInputList.add(convertedInput);
       }
 
-      traits = call.getPlanner().emptyTraitSet().plus(Prel.DRILL_PHYSICAL).plus(DrillDistributionTrait.SINGLETON);
+      // distribution trait is set to ANY to allow Union-All to inherit the distribution of its child
+      traits = call.getPlanner().emptyTraitSet().plus(Prel.DRILL_PHYSICAL).plus(DrillDistributionTrait.ANY);
```

Comment:
Makes sense. Let's put propagating from either LHS or RHS trait to downstream operator as enhancement.